src/Symfony/Component/Profiler/Profiler.php

```diff
@@ -0,0 +1,387 @@
+     *
+     * @param string $ip     The IP
+     * @param string $url    The URL
+     * @param string $limit  The maximum number of tokens to return
+     * @param string $method The request method
+     * @param string $start  The start date to search from
+     * @param string $end    The end date to search to
+     *
+     * @return array An array of tokens
+     *
+     * @see http://php.net/manual/en/datetime.formats.php for the supported date/time formats
+     * @deprecated since 2.8 and will be removed in 3.0 use Profiler::findBy instead
+     */
+    public function find($ip, $url, $limit, $method, $start, $end)
+    {
+        $criteria = array();
+        if (!empty($ip)) {
+            $criteria['ip'] = $ip;
+        }
+        if (!empty($url)) {
+            $criteria['url'] = $url;
+        }
+        if (!empty($method)) {
+            $criteria['method'] = $method;
+        }
+
+        return $this->findBy($criteria, $limit, $start, $end);
+    }
+
+    /**
+     * Profile a Request.
+     *
+     * @param Request  $request
+     * @param Response $response
+     *
+     * @return ProfileInterface|void
+     */
+    public function profileRequest(Request $request, Response $response)
+    {
+        if (!$this->enabled) {
+            return;
+        }
+
+        $profile = new HttpProfile(
+            $this->generateToken(),
+            $request->getClientIp(),
+            $request->getUri(),
+            $request->getMethod(),
+            $response->getStatusCode()
+        );
+
+        $response->headers->set('X-Debug-Token', $profile->getToken());
+
+        return $this->profile($profile, $request, $response);
+    }
+
+    /**
+     * Profile a Console Command.
+     *
+     * @param Command        $command
+     * @param InputInterface $input
+     * @param int            $exitCode
+     *
+     * @return ProfileInterface|void
+     */
+    public function profileCommand(Command $command, InputInterface $input, $exitCode)
+    {
+        if (!$this->enabled) {
+            return;
+        }
+
+        $profile = new ConsoleProfile($this->generateToken(), $command->getName(), $input->getArguments(), $input->getOptions(), $exitCode);
+
+        return $this->profile($profile);
+    }
+
+    /**
+     * Collects data.
+     *
+     * @param ProfileInterface $profile
+     *
+     * @return ProfileInterface
+     */
+    public function profile(ProfileInterface $profile, Request $request = null, Response $response = null, \Exception $exception = null)
+    {
+        foreach ($this->collectors as $collector) {
+            $collector->setToken($profile->getToken());
+            if ($collector instanceof RuntimeDataCollectorInterface) {
+                $profile->add($collector->getName(), $collector->collect());
+            } elseif ($collector instanceof LateDataCollectorInterface) {
+                // we need to clone for sub-requests
+                $profile->addCollector(clone $collector);
```

Comment:
I think we could avoid it by using the Collectors of the profiler in the `save()` method instead of the ones from the profile it self, this should effectively no longer require us to store a collector on the profile.
But I will need to check if the behaviour doesn't change as we would be remove the cloning of the collector.